src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -345,6 +521,69 @@ private static boolean advanceIteratorAndCheckLocus(final PeekableIterator<Varia
         return vcfIterator.hasNext() && CompareVariantContextToLocus(sequenceDictionary, vcfIterator.peek(), locus) == 0;
     }
 
+    /**
+     * Compares a VariantContext to a Locus providing information regarding possible overlap, or relative location
+     *
+     * @param dictionary     The {@link SAMSequenceDictionary} to use for ordering the sequences
+     * @param variantContext the {@link VariantContext} to compare
+     * @param locus          the {@link Locus} to compare
+     * @return negative if variantContext comes before locus (with no overlap)
+     * zero if variantContext and locus overlap
+     * positive if variantContext comes after locus (with no overlap)
+     * <p/>
+     * if variantContext and locus are in the same contig the return value will be the number of bases apart they are,
+     * otherwise it will be MIN_INT/MAX_INT
+     */
+    public static int CompareVariantContextToLocus(final SAMSequenceDictionary dictionary, final VariantContext variantContext, final Locus locus) {
+
+        final int indexDiff = dictionary.getSequenceIndex(variantContext.getContig()) - locus.getSequenceIndex();
+        if (indexDiff != 0) {
+            return indexDiff < 0 ? Integer.MIN_VALUE : Integer.MAX_VALUE;
+        }
+
+        //same SequenceIndex, can compare by genomic position
+        if (locus.getPosition() < variantContext.getStart())
+            return variantContext.getStart() - locus.getPosition();
+        if (locus.getPosition() > variantContext.getEnd())
+            return variantContext.getEnd() - locus.getPosition();
+        return 0;
+    }
+
+    /**
+     * If there's a variant at the locus return true, otherwise false.
+     * <p>
+     * HAS SIDE EFFECTS!!! Queries the vcfFileReader
+     *
+     * @param vcfFileReader      a {@link VCFFileReader} to query for the given locus
+     * @param locusInfo          a {@link SamLocusIterator.LocusInfo} at which to examine the variants
+     * @return true if there's a variant over the locus, false otherwise.
+     */
+    private static boolean checkLocus(final VCFFileReader vcfFileReader, final SamLocusIterator.LocusInfo locusInfo) {
+        boolean overlaps = false;
+
+        if (locusInfo != null) {
+            try (final CloseableIterator<VariantContext> vcfIterator = vcfFileReader.query(locusInfo)) {
+
+                overlaps = true;
+
+                boolean allFiltered = true;
+
+                while (vcfIterator.hasNext()) {
+                    final VariantContext vcf = vcfIterator.next();
+                    if (vcf.isFiltered()) {
+                        continue;
+                    }
+                    allFiltered = false;
```

Comment:
you can break here right?

Reply:
No - It's not guaranteed that there will be only 1 variant that overlaps the locus in the given variant file.